test/runtests.jl

```diff
@@ -1314,3 +1314,85 @@ end
         @test predict(mdl1) ≈ predict(mdl2)
     end
 end
+
+@testset "dropcollinearity in GLM" begin
+    num_rows = 100_000
+    dfrm = DataFrame()
+    dfrm[!, :x1] = randn(StableRNG(123), num_rows)
+    dfrm[!, :x2] = randn(StableRNG(456), num_rows)
+    dfrm[!, :x3]= 2*dfrm[!, :x1] + 3*dfrm[!, :x2]
+    dfrm[!, :y] = Int.(randn(StableRNG(9999999), num_rows) .> 0)
+    @testset "General test" begin
+        # an example of rank deficiency caused by linearly dependent columns
+        
+        f1 = @eval(@formula(y ~ 1+x1+x2+x3))
+
+        m1 = fit(GeneralizedLinearModel,
+                f1,
+                dfrm,
+                Binomial(),
+                GLM.LogitLink();
+                dropcollinear=true)
+        @test isa(m1.model.pp.chol, CholeskyPivoted)
+        @test rank(m1.model.pp.chol) == 3
+        @test deviance(m1) ≈ 138628.8442005168471951
+
+        f2 = @eval(@formula(y ~ 1+x1*x2*x3))
+
+        m2 = fit(GeneralizedLinearModel,
+                f2,
+                dfrm,
+                Binomial(),
+                GLM.LogitLink();
+                dropcollinear=true)
+        @test isa(m2.model.pp.chol, CholeskyPivoted)
+        @test rank(m2.model.pp.chol) == 7
+        @test deviance(m2.model) ≈ 138627.6062160052242689
+        glmallow = fit(GeneralizedLinearModel, @formula(y~x1+x2+x3), dfrm, Poisson(), dropcollinear=true)
+        @test isa(glmallow.model.pp.chol, CholeskyPivoted)
+    end
+    data = DataFrame(x1 = [4, 5, 9, 6, 5], x2 = [5, 3, 6, 7, 1], 
+                     x3=[4.2, 4.6, 8.4, 6.2, 4.2], y=[14, 14, 24, 20, 11])
+    @testset "Test with equivalent LM model" begin
+        mdl1 = lm(@formula(y ~ x1 + x2 + x3), data; dropcollinear=true)
+        mdl2 = glm(@formula(y ~ x1 + x2 + x3), data, Normal(), IdentityLink();
+                   dropcollinear=true)
+
+        @test coef(mdl1) ≈ coef(mdl2)
+        @test stderror(mdl1)[1:3] ≈ stderror(mdl2)[1:3]
```

Comment:
Also test `NaN` coefficients. Same below.

Reply:
Added `NaN` test in a test case.